test/integration/auth.test.js

```diff
@@ -254,4 +256,296 @@ describe("auth", function () {
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get("/auth/google/callback")
+      .query({ code: "codeReturnedByGoogle", state: rdsUiUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should redirect the google user to the redirect URL provided on successful login, if user has incomplete user details false", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const rdsUrl = new URL("https://dashboard.realdevsquad.com").href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: rdsUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUrl);
+  });
+
+  it("should redirect the google user to realdevsquad.com if non RDS URL provided, any url that is other than *.realdevsquad.com is invalid", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const invalidRedirectUrl = new URL("https://google.com").href;
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: invalidRedirectUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should redirect the google user to realdevsquad.com if invalid redirect URL provided", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const invalidRedirectUrl = "invalidURL";
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: invalidRedirectUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should send a cookie with JWT in the response for google user", function (done) {
```

Comment:
Please use a more appropriate name, when does this happen

Reply:
Changed the name to `should Issue Jwt Cookie On using Google Login`.